model/host/host.go

```diff
@@ -2663,6 +2663,22 @@ func (h *Host) MarkShouldExpire(expireOnValue string) error {
 	)
 }
 
+// UnsetHomeVolume disassociates a home volume from a (stopped) host.
+// This is for internal use, and should only be used on hosts that
+// will be terminated imminently; otherwise, the host will fail to boot.
+func (h *Host) UnsetHomeVolume() error {
+	err := UpdateOne(
+		bson.M{IdKey: h.Id},
+		bson.M{"$unset": bson.M{HomeVolumeIDKey: true}},
+	)
+	if err != nil {
+		return err
+	}
+
+	h.HomeVolumeID = ""
```

Comment:
I think it's generally considered okay to set this before the UpdateOne (and just return that error directly) since we return an error if it doesn't work and theoretically don't do stuff with the host anymore; we can still keep it this way if you prefer, just wanted to mention

Reply:
I usually prefer to do it this way because there are some cases where there's some continue-on-error after a DB operation fails, so you know the in-memory copy is in agreement with what's in the DB.